amr-wind/utilities/console_io.cpp

```diff
@@ -214,4 +215,75 @@ void print_tpls(std::ostream& out)
     }
 }
 
+void print_nonlinear_residual(
```

Comment:
I am not sure this print function belongs here. It feels different than the others. Not sure where to put it yet. I wonder if we can also make it more generic by templating to accept fields. The fields should also have the repo/mesh info so we don't need to pass in sim I think. Maybe this is more of a field utils kinda thing.